packages/app-extensions/src/actions/modules/actionHandlers/legacyAction.js

```diff
@@ -0,0 +1,77 @@
+import {call} from 'redux-saga/effects'
+
+export const loadScript = src => new Promise((resolve, reject) => {
+  const s = document.createElement('script')
+  s.src = src
+  s.onload = resolve
+  s.onerror = reject
+  document.head.appendChild(s)
+})
+
+export const loadCss = src => new Promise((resolve, reject) => {
+  const head = document.getElementsByTagName('head')[0]
+  const link = document.createElement('link')
+  link.rel = 'stylesheet'
+  link.type = 'text/css'
+  link.href = src
+  link.media = 'all'
+  link.onload = resolve
+  link.onerror = reject
+  head.appendChild(link)
+})
+
+export function* loadSequentially(sources) {
+  for (const source of sources) {
```

Comment:
same result but i personally prefer:

`yield all(sources.map(source => call(source.handler, source.src)))`

Reply:
not possible to use `all` here as it yields the `call` effects **in parallel**. we need to load the sources **sequentially**.